### ISSUE TITLE
Select: Change onChange event logic or introduce separate onSelect event

### ISSUE THREAD

**Issue by MatviiStelmakh** [primefaces/primeng#17076]
### Describe the bug

The ngModelChange event should be triggered on mouse click or keyboard Enter event, not the keyboard navigation.
The current behavior is not intuitive and very inconvenient if you have an action on this event like changing pagination size which leads to a page reload. Navigation is not a selection, so the model value should remain unchanged.

Similar requests were closed without implementation.
https://github.com/primefaces/primeng/issues/5335
https://github.com/primefaces/primeng/issues/9864
https://github.com/primefaces/primeng/issues/11703
https://github.com/primefaces/primeng/issues/12819

Still, for v.18, I see that a change event is emitted on keyboard navigation.

### Environment

stackblitz

### Reproducer

https://stackblitz.com/edit/angular-wzpqmh?file=src%2Fapp%2Fdemo%2Fdropdown-basic-demo.ts

### Angular version

18

### PrimeNG version

18

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

stackblitz

### Steps to reproduce the behavior

1. Open the console for observing log events
2. Open the dropdown
3. Navigate with keyboard through items (do not select)

### Expected behavior

Since navigation is not a selection, model should not change its value

I'd prefer to have the onChange event only after explicit option selection (mouse click or ENTER).

Instead of introducing a breaking change to the current logic, it will be useful to have an additional event like onSelect($event) where the event is the selected option.
Keep in mind that on leaving the dropdown (ESC, TAB, click outside...) internal state should be reset to the previous state if the user decides to leave the dropdown after some navigation.
So if it is not possible please review the possibility of changing the onChange event behavior.

**Comment by mertsincan** [comment]
Thanks a lot for your update!

**Comment by mehmetcetin01140** [comment]
fixed in v22
